apps/examples/hello/ssl_client_tr.c

```diff
@@ -0,0 +1,270 @@
+										   MBEDTLS_SSL_TRANSPORT_STREAM,
+										   MBEDTLS_SSL_PRESET_DEFAULT)) != 0) {
+		mbedtls_printf(" failed\n	 ! mbedtls_ssl_config_defaults returned %d\n\n", ret);
+		goto exit;
+	}
+
+	mbedtls_printf(" ok\n");
+
+	/* OPTIONAL is not optimal for security,
+		 * but makes interop easier in this simplified example */
+	mbedtls_ssl_conf_authmode(&conf, MBEDTLS_SSL_VERIFY_REQUIRED);
+	mbedtls_ssl_conf_ca_chain(&conf, &cacert, NULL);
+	mbedtls_ssl_conf_rng(&conf, mbedtls_ctr_drbg_random, &ctr_drbg);
+	mbedtls_ssl_conf_dbg(&conf, my_debug, stdout);
+
+	if ((ret = mbedtls_ssl_setup(&ssl, &conf)) != 0) {
+		mbedtls_printf(" failed\n	 ! mbedtls_ssl_setup returned %d\n\n", ret);
+		goto exit;
+	}
+
+	/* if ((ret = mbedtls_ssl_set_hostname(&ssl, SERVER_NAME)) != 0) { */
+	/* 	mbedtls_printf(" failed\n	 ! mbedtls_ssl_set_hostname returned %d\n\n", ret); */
+	/* 	goto exit; */
+	/* } */
+
+	mbedtls_ssl_set_bio(&ssl, &server_fd, mbedtls_net_send, mbedtls_net_recv, NULL);
+
+	/*
+		 * 4. Handshake
+		 */
+	mbedtls_printf("	. Performing the SSL/TLS handshake...");
+	fflush(stdout);
+
+	while ((ret = mbedtls_ssl_handshake(&ssl)) != 0) {
+		if (ret != MBEDTLS_ERR_SSL_WANT_READ && ret != MBEDTLS_ERR_SSL_WANT_WRITE) {
+			mbedtls_printf(" failed\n	 ! mbedtls_ssl_handshake returned -0x%x\n\n", (unsigned int)-ret);
+			goto exit;
+		}
+	}
+
+	mbedtls_printf(" ok\n");
+
+	/*
+		 * 5. Verify the server certificate
+		 */
+	mbedtls_printf("	. Verifying peer X.509 certificate...");
+
+	/* In real life, we probably want to bail out when ret != 0 */
+	if ((flags = mbedtls_ssl_get_verify_result(&ssl)) != 0) {
+		char vrfy_buf[512];
+
+		mbedtls_printf(" failed\n");
+
+		mbedtls_x509_crt_verify_info(vrfy_buf, sizeof(vrfy_buf), "	! ", flags);
+
+		mbedtls_printf("%s\n", vrfy_buf);
+	} else
+		mbedtls_printf(" ok\n");
+
+	/*
+		 * 3. Write the GET request
+		 */
+	mbedtls_printf("	> Write to server:");
+	fflush(stdout);
+
+	len = sprintf((char *)buf, GET_REQUEST);
+
+	while ((ret = mbedtls_ssl_write(&ssl, buf, len)) <= 0) {
+		if (ret != MBEDTLS_ERR_SSL_WANT_READ && ret != MBEDTLS_ERR_SSL_WANT_WRITE) {
+			mbedtls_printf(" failed\n	 ! mbedtls_ssl_write returned %d\n\n", ret);
+			goto exit;
+		}
+	}
+
+	len = ret;
+	mbedtls_printf(" %d bytes written\n\n%s", len, (char *)buf);
+
+	/*
+		 * 7. Read the HTTP response
+		 */
+	mbedtls_printf("	< Read from server:");
+	fflush(stdout);
+
+	do {
+		len = sizeof(buf) - 1;
+		memset(buf, 0, sizeof(buf));
+		ret = mbedtls_ssl_read(&ssl, buf, len);
+
+		if (ret == MBEDTLS_ERR_SSL_WANT_READ || ret == MBEDTLS_ERR_SSL_WANT_WRITE)
+			continue;
+
+		if (ret == MBEDTLS_ERR_SSL_PEER_CLOSE_NOTIFY)
+			break;
+
+		if (ret < 0) {
+			mbedtls_printf("failed\n	! mbedtls_ssl_read returned %d\n\n", ret);
+			break;
+		}
+
+		if (ret == 0) {
+			mbedtls_printf("\n\nEOF\n\n");
+			break;
+		}
+
+		len = ret;
+		mbedtls_printf(" %d bytes read\n\n%s", len, (char *)buf);
+	} while (1);
+
+	mbedtls_ssl_close_notify(&ssl);
+
+	exit_code = -1;
```

Comment:
`exit_code` is initialized from line 64, but not used anywhere.
And it is set as `-1` again, but checked it is `-1` or not in line 254.
Please check it is needed or not.